Source/NavierStokes.cpp

```diff
@@ -2498,7 +2498,7 @@ NavierStokes::getDiffusivity (MultiFab* diffusivity[BL_SPACEDIM],
 
     for (int dir = 0; dir < BL_SPACEDIM; dir++)
     {
-      diffusivity[dir]->setVal(visc_coef[diff_comp], dst_comp, 1, diffusivity[dir]->nGrow());
+      diffusivity[dir]->setVal(visc_coef[diff_comp], dst_comp, ncomp, diffusivity[dir]->nGrow());
```

Comment:
Doesn't this have to set each component to a different value from the array, so more like 

```
for (int n=0; n<ncomp; ++n) {
  diffusivity[dir]->setVal(visc_coef[diff_comp+n], dst_comp+n, 1, diffusivity[dir]->nGrow());
}
```

Reply:
To do more sophisticated diffusion, yes. But what IAMR has been doing by default is just reading a single scal_diff_coefs from inputs and setting all scalar diffusivities to that value. It's been on the user to do anything more exciting.

Reply:
I should clarify that. Temperature gets it's own diffusivity. Density and tracers use the same one.

Reply:
Ok, as long as this was intended.